### ISSUE TITLE
How does the trajectory defined in 2D image space

### ISSUE THREAD

**Issue by OpenJarvisAI** [wsakobe/TrackVLA#2]
looks the model can output tracjectory even with web videos, deos the trajectory defines in 2d image space? If so, how does the trajectory dataset generated?

**Comment by wsakobe** [comment]
Good question! In fact, we only tested the passive visual tracking capability of TrackVLA directly on web videos. The trajectories are generated by projecting the model’s output trajectories into the 2D image space using the camera intrinsics from the training data. This is mainly done to visualize the quality of the trajectory outputs.

**Comment by wsakobe** [comment]
We simply used the camera intrinsics corresponding to the training data and resized the web video frames to match the image shape of the training data, so that we could plot approximate trajectories.

**Comment by OpenJarvisAI** [comment]
does the trajectory reasonable when using another intrisics? 
also, if the raw trajectory is 3D spaces, doesn't it fitted into fixed camera position? This could be a problem when camera changed position

**Comment by wsakobe** [comment]
First, we resize the web videos to match the dimensions of the training data and use the camera intrinsics corresponding to the training data, so the two are aligned at the 2D image level. Second, since it is not possible to directly obtain depth information from a monocular camera, the trajectories learned and output by the model are necessarily related to the camera parameters. TrackVLA is trained solely on simulated data, and when transferring to a different camera, it can generate the corresponding training data in a very short time, thus enabling adaptation to different camera intrinsics.

**Comment by OpenJarvisAI** [comment]
if am understand correctly, the model output is not actually real-world 3d points, but points in camera coordinates, and mapping to realworld based on intrinsics ?

**Comment by OpenJarvisAI** [comment]
hi, is that image world 3D coordinates?

**Comment by wsakobe** [comment]
No, the trajectories are actually 3D world trajectories with respect to the robot’s coordinate frame.

**Comment by wsakobe** [comment]
The trajectory is composed of 10 waypoints, each separated by 0.1 seconds. Each waypoint is defined as (x, y, yaw). For further details, please refer to the paper. [https://arxiv.org/pdf/2505.23189](url)

**Comment by wsakobe** [comment]
When deploying in the real world, the position of the robot’s camera should be aligned with that used in the simulation data as closely as possible in order to minimize the sim-to-real gap.

**Comment by wsakobe** [comment]
In fact, you can quickly collect simulation data and retrain the corresponding model based on the parameters of the camera mounted on your robot, which is a key advantage of TrackVLA’s use of purely simulated data.
